modules/seqexec/server/src/main/scala/seqexec/server/InstrumentControllerSim.scala

```diff
@@ -119,4 +121,10 @@ class InstrumentControllerSim(name: String, useTimeout: Boolean) {
 object InstrumentControllerSim {
   def apply(name: String): InstrumentControllerSim = new InstrumentControllerSim(name, false)
   def withTimeout(name: String): InstrumentControllerSim = new InstrumentControllerSim(name, true)
+
+  implicit class SimulationExceptionOps[F[_]: Monad, A](s: EitherT[F, SeqexecFailure, A]) {
+    def orSimulationError(implicit ev: ApplicativeError[F, SeqexecFailure]): F[A] =
+      s.getOrElseF(ev.raiseError(SeqexecFailure.FailedSimulation))
```

Comment:
This will discard any existing error in the `EitherT` … is this your intent? If you want to absorb the existing error into `F` you can say `s.value.flatMap(_.liftTo[F])` … if not maybe at least log it?

Reply:
Good question. I think I need to revisit this. I'm still finding my way around `AppicativeError`. I was less concerned as the simulation never fails and this is explicitly for simulations